Release: v2.4.1

attackant opened the pull request:
Feature branch for [the v2.4.1 release](https://github.com/alleyinteractive/apple-news/milestone/21).

Commit:
Date data should be in UTC before passing to strtotime()

PHP's `strtotime()` uses the epoch time which is expecting the time format in UTC instead of local time. 

This PR updates `$date` to use `$post->post_date_gmt` instead of `$post->post_date` for proper timezone conversations. 

Fixes #1006

mrmadhat committed:
Use options.php instead of null

attackant committed:
Refactor node-tests workflow with a common GitHub Action

The node-tests workflow steps have been replaced with a shared GitHub Action from 'alleyinteractive'. This simplifies the workflow file making it more readable and maintainable. Additional configurations like ci mode, node version, and working directory are enabled; hence enhancing the node-tests setup process.

attackant committed:
Refactor PHPUnit GitHub workflow for simplicity

The PHPUnit GitHub workflow was significantly simplified by reducing the amount of work done in the YAML file. The previous workflow was manually setting up the test environment (downloading WordPress, setting up MySQL, etc.) within this file.

The updated workflow uses a custom GitHub action that handles these setup steps, resulting in a cleaner and easier to read workflow file.

Moreover, the PHP versions and WordPress version for testing have been updated to reflect more current versions.

attackant committed:
"Update composer script for testing

Added new script 'test' to composer.json file. This script combines two existing scripts, 'phpcs' and 'phpunit', making it more efficient to run both scripts for testing. Also, 'phpcs' script has been updated with '--basepath=.' argument to set the base directory for relative paths."

attackant committed:
Update phpcs command in composer.json

The basepath option was removed from the phpcs command in composer.json. The basepath option was found to be unnecessary because phpcs runs relative to the project root by default. Thus, removing this option simplifies the command without changing its functionality.

attackant committed:
Update composer dependencies and PHP version

This commit updates the composer.json file. The dependencies versions have been changed to more recent versions and the required PHP version has been updated too. These changes support more recent and secure code usage conventions.

attackant committed:
Simplify PHPCS action in workflow file

Refactored the PHP Coding Standards action block in Github workflow file (.github/workflows/phpcs.yml). Removed redundant steps, and migrated job definition to use the prepared PHP coding standards action. This change provides a cleaner and less error-prone setup, allowing more focus on PHP Coding Standards execution.

attackant committed:
Remove WordPress matrix from PHPCS workflow

Removed the WordPress version matrix from the PHP Code Sniffer (PHPCS) GitHub Actions workflow. This is because we no longer need to test against different WordPress versions, simplifying the workflow and reducing the resources needed for testing.

attackant committed:
Remove redundant "phpcs" command from workflow

The "command: phpcs" line was removed from the phpcs.yml workflow file. This change was necessary as the command is already specified in the php-coding-standards.yml workflow from alleyinteractive repository and doesn’t need to be defined twice. This refactoring helps to avoid confusion and potential command conflicts in the future.

attackant committed:
Add null checks and private variables for to address PHP notices and deprecations

Added null checks in `function-coauthors.php` and `class-components.php` to prevent possible undefined behaviour when handling non-initialized variables. This makes the code more robust and resilient to potential errors. Moreover, added private string variables in `test-class-table.php` and private `Admin_Apple_Themes` variable to the `test-class-admin-apple-themes.php` which improves the readability and structure of these files.

renatonascalves commented on composer.json:
```diff
@@ -5,18 +5,22 @@
-    "composer/installers": "~1.0",
-    "php": ">=7.1"
+    "composer/installers": "^1.12.0",
+    "php": "^8.0"
```
Odd: in the Github Action matrix, you are testing for PHP 7.4 too, but here you are requiring 8.0+. 🤔

kevinfodness replied:
Yeah, this should be updated to >= 7.4

attackant replied:
Fixed.

renatonascalves commented on includes/apple-exporter/builders/class-components.php:
```diff
@@ -526,6 +526,10 @@ private function get_components_from_node( $node ) {
 	 */
 	private function get_image_full_size_url( $url ) {
 
+		if(null === $url) {
```
```suggestion
		if( empty( $url ) ) {
```

kevinfodness replied:
Additionally, this is going to fail phpcs checks because there needs to be a space after the `if`. You should be running `composer phpcs` locally and ensuring it is clean before creating a PR, and should ensure that all of the automated checks pass before putting it in for code review.

attackant replied:
Noted, I didn't want to run phpcbf yet because so many lines of code were going to be impacted, and I wanted to get early feedback, but will do before I finalize this PR. Thank you.

renatonascalves commented on tests/mocks/function-coauthors.php:
```diff
@@ -29,6 +29,10 @@ function coauthors( $between = ', ', $between_last = ' and ', $before = '', $aft
 	// Get last index.
 	$last_index = count( $apple_news_coauthors ) - 1;
 
+	if(null === $between) {
```
```suggestion
	if( empty( $between ) ) {
```

renatonascalves commented on .github/workflows/node-tests.yml:
```diff
@@ -4,25 +4,10 @@ on:
+    uses: alleyinteractive/.github/.github/workflows/node-tests.yml@main
+    with:
+      ci: true
+      node: 16
+      run-audit: false
```
Those are already the default values.
```suggestion
```

kevinfodness replied:
However, the root workflow could be updated to have different defaults, which could break this workflow. I'm fine with having them be explicitly set here, even if it duplicates the current defaults.

renatonascalves replied:
Good point!

renatonascalves commented on .github/workflows/phpcs.yml:
```diff
@@ -1,25 +1,13 @@
+    strategy:
+      matrix:
+        php: [ 8.0 ]
+    uses: alleyinteractive/.github/.github/workflows/php-coding-standards.yml@main
```
Suggestion: https://github.com/alleyinteractive/.github?tab=readme-ov-file#php-composer-script

To use both `phpcs` and `phpunit` in a single action.

kevinfodness replied:
That's not how we have our reusable workflows set up currently. We might move to that at some point in the future, but that's out of scope for this update.

renatonascalves reviewed (dismissed):
a few suggestions

kevinfodness commented on .github/workflows/phpcs.yml:
```diff
@@ -1,25 +1,13 @@
 name: PHP Coding Standards
+
 on:
   pull_request:
```
All of these PR test workflows need to be updated to include the pull request branch restriction (only on develop) plus the `types` filter and the `if` statement under `jobs` as seen here: https://github.com/alleyinteractive/create-wordpress-project/blob/develop/ci-templates/.github/workflows/coding-standards.yml. Please also include the (uncommented) schedule block as well.

attackant replied:
Got it thank you!

kevinfodness commented on .github/workflows/phpcs.yml:
```diff
@@ -1,25 +1,13 @@
-        run: composer phpcs
+    strategy:
+      matrix:
+        php: [ 8.0 ]
```
Might as well run this under 8.2

kevinfodness commented on .github/workflows/phpunit.yml:
```diff
@@ -1,51 +1,16 @@
     strategy:
-      fail-fast: true
       matrix:
-        multisite: [0, 1]
```
Need to include the multisite variable in the new matrix, and need to pass it to the action using `with`. We need to test this plugin for use on multisite and single site installations.

kevinfodness commented on tests/admin/test-class-admin-apple-themes.php:
```diff
@@ -16,6 +16,7 @@
  * A class which is used to test the Admin_Apple_Themes class.
  */
 class Admin_Apple_Themes_Test extends Apple_News_Testcase {
+	private Admin_Apple_Themes $themes;
```
I'm sure this is due to dynamic properties. Adding these is the right call, but you need to add phpdoc to document them as well.

kevinfodness commented on tests/apple-exporter/components/test-class-table.php:
```diff
@@ -18,6 +18,8 @@
  * @subpackage Tests
  */
 class Apple_News_Table_Test extends Apple_News_Component_TestCase {
+	private string $html;
```
Same here, and same with any added properties anywhere else in this PR

attackant committed:
Downgrade PHP version requirement in composer.json

PHP version requirement in composer.json was changed from "^8.0" to "^7.4" to ensure compatibility with older PHP environments. This will allow the project to be integrated into systems that have not yet updated to the newest PHP version.

attackant committed:
"Configure PHP CodeSniffer to run only in non-draft PRs"

This update adjusts the GitHub Action for running PHP CodeSniffer to trigger only for non-draft pull requests on 'develop' branch. The purpose is to minimize unnecessary runs and provide feedback only when code is ready for review.

attackant committed:
Update GitHub Actions to refine pull request triggers

This commit updates the GitHub workflow for PHPunit, PHPCS, and node tests. The workflows are now triggered only pull requests towards the develop branch and exclude draft pull requests. This prevents unnecessary running of these workflows on every pull request. Additionally, the PHP version for PHPCS has been updated from 8.0 to 8.2 to align with the latest PHP version usage.

attackant committed:
Add multisite testing to GitHub Actions

An addition to the test matrix in the GitHub Actions configuration for PHPUnit was made to include multisite setups. This change ensures that the codebase is compatible with WordPress multisite environments.

attackant committed:
Add PHP docblocks for HTML and caption fields in Table_Test and Themes_Test classes

attackant committed:
Fix failing unit tests due to unset defaults

Amended default conditions from null checks to empty checks in function-coauthors.php and class-components.php. Previously, there were no defaults set when running unit tests which led to unexpected fails. This change will ensure that unit tests pass even when certain parameters are not explicitly set.

attackant committed:
"PHPCBF changes updated class imports and formatting across multiple modules"

Removed unnecessary aliases from class imports for improved readability and simplified access statements. Change usage of 'dirname( FILE )' to '__DIR__' for cleaner calls. Also made minor formatting updates, such as making spaces consistent around parentheses and adding increment from prefix to postfix form in for loops for better standardization. The changes were done to increase the code quality without affecting the functional behavior of the codebase.

attackant committed:
Update PHP Code Sniffer suppression annotations

This commit updates the PHP Code Sniffer suppression annotations in the apple-exporter components class. Previously, there was an incorrect ignore annotation `VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable` in `node_matches` method which has now been replaced with `Generic.CodeAnalysis.UnusedFunctionParameter.Found`. This change better reflects the actual warning being suppressed.

Also, a new annotation for `WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase` has been added around the method for finding iframe nodes in

attackant committed:
Ignore specific PHP Code Sniffer warnings for audio node checks

This change was necessary to ensure the codebase conforms to the WordPress Naming Conventions. Comments were added to ignore PHP Code Sniffer warnings about non-snake case variables when checking if a node is an 'audio' node. This had to be done since those checks are part of the application's logic

attackant committed:
"Escaping response messages and adding PHPCS exceptions"

In this commit, we have made several changes to ensure safe output and to follow proper PHP coding standards.
1. We have started using 'esc_html' function to escape the outputs of the exception messages in the 'class-request.php' file. This is to prevent potential security leaks from unescaped outputs.
2. We also added comments to ignore phpcs warnings for certain lines which were raising warnings due to variable naming issues. Especially for lines using `nodeName` and `keyPath` properties, as they do not follow WordPress's snake case convention, but they are necessary for our function to work correctly. Further, they are properties of node, which we do not have control over, hence

attackant committed:
Ensure proper escaping of error messages in class-push.php

Escaping functions have been added to all error messages in class-push.php. This change is made to prevent potential Cross-Site Scripting (XSS) vulnerabilities by ensuring any user-supplied data is properly escaped before it is output, following best practices for WordPress development. This does not affect the functionality of the error messages but enhances the security.

attackant committed:
"Adjusted code to satisfy PHPCS naming conventions

Code was updated across video, tweet, and embed-web-video classes to align with PHP Code Sniffer's WordPress naming conventions. Ignored variable checks were placed on non-snake cased node properties, and text strings were escaped in delete and meta classes. Aimed to enhance code quality and prevent potential scripting attacks."

attackant committed:
"Add PHP Code Sniffer ignores for snake case naming checks

The project has been updated to add comments to ignore the 'not using snake case' warnings from PHP Code Sniffer. This was necessary as the property names used in the Node class provided by the core PHP library, do not follow snake case convention. These warnings were leading to unnecessary noise in the linting reports and were misleading as it could lead developers to think that there's something wrong with the project's own codebase. As this convention is not something developer can change, it is now simply ignored by the linter."

attackant committed:
Update trigger event location in GitHub workflows

This commit repositions the list of trigger event types in .github/workflows/phpunit.yml, .github/workflows/phpcs.yml, and .github/workflows/node-tests.yml from under "branches" to directly under "pull_request". This modification accurately reflects the intended YAML syntax, ensuring that the workflows are triggered properly for the specified pull request events, namely

attackant committed:
Update PHP version requirement in composer.json

Changed the PHP version requirement from "^7.4" to ">=7.4" in the composer.json file. This was necessary to allow the software to run in environments with PHP 8 or higher, increasing its compatibility and usability.

attackant committed:
Change submenu parent for Bulk Export page to handle PHP deprecation notices.

In this commit, the Bulk Export submenu in the apple-news plugin was changed to not appear in any menu. The parent of the submenu was previously null, but it has now been set to 'options.php'. This change is intended to provide more organization and decrease clutter in the main menu.

attackant committed:
Merge branch 'mrmadhat-chore/php-8-depreciations' into release/v2.4.1

mogmarsh commented on includes/apple-exporter/components/class-embed-generic.php:
```diff
@@ -51,6 +51,7 @@ public static function node_matches( $node ) {
 			&& 'iframe' === $node->childNodes->item( 0 )->nodeName
 		) {
 			return $node;
+			/* phpcs:enable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase */
```
it probably works fine as is, but I feel like the enable shouldn't be inside the `if`... my mind wants it to be at the same level as the disable.

mogmarsh commented on includes/apple-exporter/components/class-facebook.php:
```diff
@@ -105,6 +106,7 @@ public static function node_matches( $node ) {
 			) {
 				return $node;
 			}
+			/* phpcs:enable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase */
```
same

mogmarsh commented on includes/apple-exporter/components/class-tiktok.php:
```diff
@@ -45,6 +46,7 @@ public static function node_matches( $node ) {
 			&& self::node_has_class( $node, 'tiktok-embed' )
 		) {
 			return $node;
+			/* phpcs:enable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase */
```
same

mogmarsh commented on tests/apple-exporter/components/test-class-table.php:
```diff
@@ -18,6 +18,19 @@
 class Apple_News_Table_Test extends Apple_News_Component_TestCase {
+	/**
+	 * The HTML code to display
+	 * .
```
```suggestion
```

mogmarsh commented on admin/apple-actions/index/class-push.php:
```diff
@@ -227,8 +227,8 @@ private function push( $user_id = null ) {
-					__( 'Skipped push of article %d due to the apple_news_skip_push filter.', 'apple-news' ),
-					$this->id
+					esc_html__( 'Skipped push of article %d due to the apple_news_skip_push filter.', 'apple-news' ),
+					$this->id // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
```
any reason not to escape this?

mogmarsh commented on admin/apple-actions/index/class-push.php:
```diff
@@ -277,8 +277,8 @@ private function push( $user_id = null ) {
-						__( 'Skipped push of article %d due to the presence of a skip push taxonomy term.', 'apple-news' ),
-						$this->id
+						esc_html__( 'Skipped push of article %d due to the presence of a skip push taxonomy term.', 'apple-news' ),
+						$this->id // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
```
any reason not to escape this?

mogmarsh commented on admin/apple-actions/index/class-push.php:
```diff
@@ -393,8 +393,8 @@ private function push( $user_id = null ) {
-					__( 'Skipped push of article %d to Apple News because it is already in sync.', 'apple-news' ),
-					$this->id
+					esc_html__( 'Skipped push of article %d to Apple News because it is already in sync.', 'apple-news' ),
+					$this->id // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
```
any reason not to escape this?

mogmarsh commented on includes/apple-push-api/class-mime-builder.php:
```diff
@@ -210,8 +210,8 @@ private function build_attachment( $name, $filename, $content, $mime_type, $size
-					__( 'The attachment %s could not be included in the request because it was empty.', 'apple-news' ),
-					esc_html( $filename )
+					esc_html__( 'The attachment %s could not be included in the request because it was empty.', 'apple-news' ),
+					$filename // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
```
any reason not to escape this?

mogmarsh commented on includes/apple-push-api/class-mime-builder.php:
```diff
@@ -221,9 +221,9 @@ private function build_attachment( $name, $filename, $content, $mime_type, $size
-					esc_html( $filename ),
-					esc_html( $size )
+					esc_html__( 'The attachment %1$s could not be included in the request because its size was %2$s.', 'apple-news' ),
+					$filename, // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
+					$size // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
```
any reason not to escape this?

mogmarsh reviewed (approved):
there is nothing blocking here. I left a few little comments, do with them as you wish. 👍

attackant committed:
Improve output sanitization across various files

The commit includes adjustments in several files to improve the way outputs are sanitized, enhancing the security of the application. Escaping functions were applied to the $filename and $size parameters in class-mime-builder.php, and to $this->id in class-push.php. In addition, minor corrections were made in phpcs comments in class-embed-generic.php and class-tiktok.php, ensuring that they deactivate the corresponding sniffs after the required code, as intended.

attackant committed:
build

attackant committed:
build

attackant committed:
Update version requirements and contributors in readme

The readme file has been updated to reflect the new minimum requirements for the software: WordPress version was raised to 5.2, PHP version is now 7.4. A new contributor was added to the list and a revision in the updates history, specifically on version 2.0.0, was made. These changes are primarily to maintain the accuracy and relevance of the information in the readme file.

attackant committed:
Fix typo

attackant committed:
Temporarily add this branch

attackant committed:
Fix syntax

attackant committed:
Force running actions

attackant committed:
Add workflow_dispatch

attackant committed:
build

attackant committed:
Add workflow_dispatch and remove dummy function

attackant committed:
Testing new GH action setup PHPUnit Testing Suite workflow

attackant committed:
Update PHPUnit workflow for better test environment setup

Changed the .github/workflows/phpunit.yml workflow to improve the setup of the test environment. Instead of directly running tests, an additional step is included to creating a specific directory for WordPress PHPUnit tests. This provides a more separated and clean environment for running tests, potentially eliminating conflicts or issues with other tests or processes.

attackant committed:
Update 'composer.json' dependencies and config

Add 'mantle-framework/testkit' dependency and allow 'alleyinteractive/composer-wordpress-autoloader' in the config. The 'mantle-framework/testkit' dependency was necessary for the better testing environment setup. Updated the order of 'phpspec/prophecy' for consistency, and added 'alleyinteractive/composer-wordpress-autoloader' to the plugin allowances to ensure the autoloading of WordPress installations via Composer.

attackant committed:
Refactor bootstrap test file to improve code organization and readability

The bootstrap test file was restructured to enhance its readability and organization. A function, `apple_news_require_file`, was added to check the existence of a file before requiring it, thereby centralizing error handling. The loading of the plugin was also adjusted to use the `apple_news_require_file` function and provide a better structure. Furthermore, the mocked Brightcove functionality activation was integrated into the loading process. The changes should make the file easier to maintain and understand.

attackant committed:
Testing on PHP 8.0

attackant committed:
Testing on PHP 8.0

attackant committed:
Update autoload path and plugin loading sequence in test bootstrap

This commit updates two main aspects in the test bootstrap file. Firstly, it modifies the path of the autoload file required for prophecy from using __DIR__ to using __FILE__. This change is done to ensure correct file path resolution in different environments.
Secondly, the loading sequence of the apple-news.php plugin has been changed. It is now loaded inside the BC_Setup initialization closure, guaranteeing its availability before the 'install' method execution. This adjustment ensures the correct plugin loading

attackant committed:
Refactor directory traversal in bootstrap script

Refactored the way directory navigation is handled in the bootstrap script by replacing `dirname( dirname( __FILE__ ) )` with the more simplified `dirname( __DIR__, 1 )`. This change simplifies the code and improves readability. It does not change the functionality of the application.

attackant committed:
Add alleyinteractive/composer-wordpress-autoloader to composer.json

We are introducing the use of the `alleyinteractive/composer-wordpress-autoloader` plugin. This plugin enables us to manage and autoload WordPress package dependencies via Composer.

attackant committed:
Merge branch 'refs/heads/release/v2.4.1' into issue-1021

attackant committed:
Remove temporary PHPUnit Testing Suite, adjust composer.json and modify bootstrap.php for tests

Deleted the temporary .github/workflows/php-tests-temp.yml file as it is no longer needed. Adjusted composer.json to eliminate a duplicated plugin. In tests/bootstrap.php, moved some code blocks inside the function and also set the WP_TESTS_MULTISITE constant to 1. This cleanup and reorganization improves the readability and efficiency of code and aligns with testing requirements.

attackant committed:
"Reformat WP_TESTS_MULTISITE definition in bootstrap.php"

This commit adjusts the formatting of the WP_TESTS_MULTISITE constant definition in bootstrap.php. The constant declaration was moved onto its own line to improve the legibility of the code. Unnecessary whitespace was also removed. The change does not affect the functionality of the codebase.

attackant committed:
Update test-class-end-of-article data sets

The data sets for 'cover', 'slug', 'title', and 'byline' and 'cover', 'slug', 'title', 'author', and 'date' in test-class-end-of-article.php were updated for Mantle Testkit. The second and third values for each data set were incremented by one for accurate test results.

attackant committed:
Update minimum required PHP and WordPress versions

Updated the minimum required PHP version from 7.4 to 8.0, and the minimum required WordPress version from 5.2 to 5.6. This was done to ensure compatibility with the new enhancements and deprecations included in the 2.4.1 update of the project. The updated environments offer better performance, security and support for the latest functions.

attackant committed:
Merge pull request #1023 from alleyinteractive/issue-1021

Fix phpunit bootstrapping issues

attackant committed:
Add workflow to deploy plugin to WordPress.org

This commit adds a new GitHub action workflow `deploy-to-wordpress.yml`. This workflow automates the plugin deployment to WordPress.org repository whenever a new tag is pushed to the develop branch. It includes all the necessary steps: setting up Node and PHP environments, installing dependencies, building the project, creating a GitHub release, creating and uploading plugin ZIP archive, and finally deploying the plugin to WordPress. This automation will significantly reduce human error and speed up the plugin release process.

attackant committed:
Update .github/workflows/deploy-to-wordpress.yml

Co-authored-by: Renato Alves <19148962+renatonascalves@users.noreply.github.com>

attackant committed:
Update deployment workflow for version checking

The ".github/workflows/deploy-to-wordpress.yml" file has been modified to improve version control management. Changes include the addition of steps to get the current version, retrieve the latest tag, and check if the version has been incremented. These checks ensure that a new release has an incremented version number, preventing overwriting of previous releases. Removed the redundant "Get

attackant committed:
Update GitHub Action to abort on non-incremented versions

The GitHub Action workflow which deploys to WordPress is updated to exit with an error if the version is not incremented compared to the previous one. This change is introduced to prevent unintentional overwrites of existing plugin versions on WordPress. New 'Build' and 'Publish' steps will run only when version increment is detected, ensuring that each published version is unique.

attackant committed:
Update GitHub Actions workflow for better efficiency

This commit moves the GITHUB_TOKEN to the environment section of the release job and removes it from individual steps for better readability. It also improves the version comparison condition for more accuracy, and adds condition checks for each release-related task to only run if the version has incremented. This reduces unnecessary job executions, leading to quicker and more efficient workflow runs.

attackant committed:
add dry run

attackant committed:
add dry run

attackant committed:
Refactor deploy-to-wordpress workflow and update composer.json

This commit makes a major clean-up of the .github/workflows/deploy-to-wordpress.yml file. Now, it uses the 'alleyinteractive/.github/.github/workflows/built-release.yml' for building a draft release. It removes redundant steps that were repetitively checking for version increments and carrying out operations like setting up Node and PHP, build, etc., only if versions had been incremented. It allows composing the archive excluding specified files and directories. These changes simplify the configuration and make the process more efficient. Also, composer.json has been updated to exclude unnecessary files and directories when creating the archive.

attackant committed:
Add .distignore and update .deployignore

Added a new .distignore file and updated .deployignore to exclude files and directories we don't want included in the WordPress.org distribution or deployment. These changes aim to keep our deployment clean and free from unnecessary or unrelated files, such as configuration files for various developer tools or temporary/cache files created by the operating system or development environment.

attackant committed:
Move GITHUB_TOKEN environment variable to specific step

The GITHUB_TOKEN environment variable has been moved from the global environment of the workflow to the specific step where it is needed. This improves environment variable management and reduces unnecessary exposure of the GITHUB_TOKEN.

attackant committed:
Add built-release workflow and update deployment workflow

Added a new GitHub Actions workflow for building releases (`built-release.yml`). This workflow triggers on push events to `develop`, `main`, and `production` branches. It uses a predefined workflow from the `alleyinteractive` GitHub org, setting Node.js and PHP versions and enabling Composer installation.

Furthermore, the existing deployment to WordPress.org workflow (`deploy-to-wordpress.yml`) has been simplified. It no longer manually creates

attackant committed:
Update package.json version and script

The version of the 'publish-to-apple-news' package has been updated from "2.4.0" to "2.4.1". Additionally, two new scripts 'prerelease' and 'release' were added. The 'prerelease' script checks the node version, and the 'release' script creates a new release. These changes are made to facilitate the release process and ensure consistent node version during pre-release.

attackant committed:
Add release instructions to README

This commit adds a new section to the README file, detailing the steps required to release the plugin. It includes information on how the plugin utilises a built release workflow to compile and tag releases, as well as instructions for initiating a new release. This addition aims to provide clarity for future contributors or maintainers of the project.

attackant committed:
Remove Github actions workflows

Delete build.yml and release.yml which were previously used for managing automated builds and releases via Github Actions.

These workflows were removed because they are no longer in use. The automated building and releasing of the application is now handled by a separate CI/CD system. As such, these workflows are redundant and potentially confusing to future contributors.

attackant committed:
Update plugin version in apple-news.php

The version number of the 'Publish to Apple News' plugin has been updated from 2.4.0 to 2.4.1. This was done to reflect the latest changes or improvements made in the plugin.

attackant committed:
Merge pull request #1025 from alleyinteractive/deploy-to-wordpress-via-github

Add workflow to deploy plugin to WordPress.org

attackant committed:
Merge pull request #1007 from BrookeDot/patch-1

Date data should be in UTC before passing to strtotime()

srtfisher committed:
Bumping version in apple news class

srtfisher committed:
Remove polyfills reference (not in use with testkit)

srtfisher committed:
Ensure that the request class is loaded

srtfisher committed:
Use the WpOrg\Requests\Utility\CaseInsensitiveDictionary class if it exists

In 6.4, WpOrg\Requests\Utility\CaseInsensitiveDictionary was introduced and using Requests_Utility_CaseInsensitiveDictionary will throw an exception

srtfisher committed:
Disable nonce checking when testing

srtfisher committed:
Testing CI

srtfisher committed:
Remove workflow_dispatch

srtfisher committed:
More remove workflow_dispatch

srtfisher committed:
Adding more branches to CI

srtfisher committed:
Fix command

srtfisher committed:
Enable multisite via the matrix

srtfisher committed:
Include push on branches too

srtfisher committed:
Fix matrix variable

attackant committed:
Merge pull request #1029 from alleyinteractive/hotfix/testkit-fixes

Fix: Get Unit tests passing for 2.4.1

attackant committed:
Decode HTML entities in link button text to prevent things like ampersands from being displayed as entities

The update made in class-link-button.php fixes an issue with HTML entities being displayed in the link button text. By using the 'html_entity_decode' function, these HTML entities are now properly decoded, improving the visual output and user experience.

attackant committed:
phpcbf fixes only

attackant committed:
Update package dependencies with npm audit fix

This update upgrades the versions of several dependencies in package-lock.json. Affected packages include "@babel/code-frame" and "@babel/highlight", as well as several related packages. Also, multiple packages had their "resolved" URLs and "integrity" hashes updated to match their new version numbers.

attackant committed:
Update custom property name in .phpcs.xml

Changes were made in .phpcs.xml file to rename the custom properties whitelist rule to allowed_custom_properties

attackant committed:
Update PHP and WordPress version requirements

Updated the minimum required PHP version to 8.0 and WordPress version to 6.3 in .phpcs.xml and readme.txt. This ensures compatibility with newer versions of PHP and WordPress, and reflects requirement changes in the project setup.

attackant committed:
Fix formatting in class-markdown.php

Corrected the syntax for incrementing the list_index in class-markdown.php. The updated code ensures that the variable is incremented in a clear and well-recognized format, improving code readability and maintaining consistency across the project.

attackant committed:
Update 'Tested up to' version in readme.txt

The 'Tested up to' version information in readme.txt has been updated from version 6.3.2 to 6.4.2. This change reflects the latest version that the software has been confirmed as compatible with, providing users with updated and accurate information.

renatonascalves commented on .phpcs.xml:
```diff
@@ -69,5 +69,5 @@
 
   <!-- The version set here matches the minimum version in readme.txt. -->
-  <config name="minimum_supported_wp_version" value="4.0" />
+  <config name="minimum_supported_wp_version" value="6.3" />
```
This changed!

```suggestion
  <config name="minimum_wp_version" value="6.3" />
```

attackant committed:
Update .phpcs.xml

Co-authored-by: Renato Alves <19148962+renatonascalves@users.noreply.github.com>

attackant committed:
Refactor function parameters and condition in apple-news.php

The parameters in the function "filter_update_post_metadata" have been refactored for better readability and stricter typing. In addition, the condition to check the count of 'old_value' has been enhanced to prevent potential errors when 'old_value' is not an array.

attackant committed:
Removed strict typing in "filter_update_post_metadata" function

The strict typing in the parameters of the function "filter_update_post_metadata" has been removed to offer more flexibility. Simultaneously, the condition which checks the count of 'old_value' has been modified to avoid errors if 'old_value' is not an array.

attackant committed:
Add space

attackant committed:
Merge pull request #1037 from alleyinteractive/issue-1034

Fixes #1034

attackant committed:
Remove draft from actions
